lib/kubernetes-deploy/runner.rb

```diff
@@ -295,25 +296,38 @@ def validate_configuration
       KubernetesDeploy.logger.info("All required parameters and files are present")
     end
 
-    def update_tprs(resources)
-      resources.each do |r|
+    def deploy_resources(resources, prune: false)
+      KubernetesDeploy.logger.info("Deploying resources:")
+
+      # Apply can be done in one large batch, the rest have to be done individually
+      applyables, individuals = resources.partition { |r| r.deploy_method == :apply }
+
+      individuals.each do |r|
         KubernetesDeploy.logger.info("- #{r.id}")
         r.deploy_started = Time.now.utc
-        _, _, st = run_kubectl("replace", "-f", r.file.path)
+        case r.deploy_method
+        when :replace
+          _, _, st = run_kubectl("replace", "-f", r.file.path)
+        when :replace_force
+          _, _, st = run_kubectl("replace", "--force", "-f", r.file.path)
+        end
 
         unless st.success?
           # it doesn't exist so we can't replace it
-          run_kubectl("create", "-f", r.file.path)
+          _, err, st = run_kubectl("create", "-f", r.file.path)
+          unless st.success?
+            raise FatalDeploymentError, <<~MSG
+              Failed to replace or create resource: #{r.id}
+              #{err}
+            MSG
+          end
         end
       end
-    end
 
-    def deploy_resources(resources, prune: false)
-      KubernetesDeploy.logger.info("Deploying resources:")
+      apply_all(applyables, prune)
+    end
 
-      # TPRs must use update for now: https://github.com/kubernetes/kubernetes/issues/39906
```

Comment:
Can you move this note to above the `tpr? ? :replace : :apply` line? We will want to change TPRs to `apply` once all our clusters are on 1.6 because these resources otherwise can't get pruned.